packages/itwin/tree-widget/src/components/trees/VisibilityTreeEventHandler.ts

```diff
@@ -139,24 +139,24 @@ export class VisibilityTreeEventHandler extends UnifiedSelectionTreeEventHandler
     };
 
     new Observable<CheckboxStateChange[]>((subscriber) => event.stateChanges.subscribe(subscriber))
-      .pipe(
-        mergeMap((changes) => this.changeVisibility(changes)),
-      )
+      .pipe(mergeMap((changes) => this.changeVisibility(changes)))
       .subscribe({
+        next: (nodeId) => {
+          void this.updateCheckboxes([nodeId]);
```

Comment:
Do we need to call `updateCheckBoxes()` from `handleStateChanged` if we do that for each node individually?

Reply:
This call updates only checkbox of the node specified by `nodeId`. But I believe other nodes might be impacted also. So calling `updateCheckboxes` on complete makes sure that we update all checkboxes based on viewport state after all changes.